tock/tock/static/js/components/save_csv.js

```diff
@@ -1,8 +1,34 @@
+// Wait until the DOM content is loaded. In testing, it seems this script
+// can be loaded before the DOM is populated, so... wait! In our case, the
+// buttons remain in the DOM forever once they're loaded. This approach does
+// not work if the buttons are removed from the DOM when hidden, but the
+// accordions we have just hide their content when collapsed rather than
+// remove it.
+window.addEventListener('DOMContentLoaded', function () {
+    var downloadButtons = document.querySelectorAll('button[data-csv]');
+
+    for (var i = 0; i < downloadButtons.length; i += 1) {
+        var button = downloadButtons.item(i);
+
+        // Use an IIFE to capture variables. Otherwise all of the button events
+        // will use the values from the last button, which would be wrong. A
+        // classic Javascript quirk...
+        (function () {
+            var table_id = button.getAttribute('data-csv-id');
+            var filename_prefix = button.getAttribute('data-csv-name');
+
+            button.addEventListener('click', function () {
+                download_table_as_csv(table_id + "-table", filename_prefix);
+            });
+        })();
+    }
+});
+
 // Quick and simple export target #table_id into a csv
 // inspired by https://stackoverflow.com/a/56370447
 function download_table_as_csv(table_id, filename_prefix) {
-	var csv_string = table_to_csv(table_id);
```

Comment:
The Javascript changes from here down are where I ran `prettier` on the code. I tried to configure prettier to match the existing style instead of introducing a bunch of whitespace changes. It seems like most of the changes are whitespace alignment or semicolons.